### ISSUE TITLE
toJson() doesn't serialize all properties of derived Actions

### ISSUE THREAD

**Issue by kuhnma-prv** [poychang/Message-Card-Model#4]
The following example misses the "target" properties for OpenUriAction in the serialized json string 
`var msgCard = new MessageCard
{
  Title = "Message Title (optional)",
  Text = "This is a sing line **message** with markdown",
  Sections = new []
  {
    new Section
    {
      Text = "Yet another block with an action",
      Actions = new OpenUriAction[]
      {
        new OpenUriAction
        {
          Name = "My OpenUri action",
          Type = ActionType.OpenUri,
          Targets = new []
          {
            new Target { Uri = "https://somesite.com/xxx" },
          }
        },
      }
    },
  },
};
`
this is problably because the Actions enumeration is only serialied as IAction and not with the derived data type

**Comment by poychang** [comment]
System.Text.Json and Json.NET have different behavior. Json.NET works fine with your scenario. I'll try to figure out how to use System.Text.Json to fit this issue.

**Comment by MiguelMinoldo** [comment]
I'm facing the exact same issue, I saw in the code that you've fixed this issue adding a custom action ActionConverter, but this is not in Nuget yet.
Thanks!

**Comment by poychang** [comment]
@MiguelMinoldo I published 1.1.2  you can try it now.

**Comment by willl** [comment]
@poychang I'm using 1.1.2 and am having issues with ToJson() and Actions.

Here's some sample code for the card generation

```C#
var card = new MessageCard
{
    Title = "This is the title",
    Text = "Message text",
    Sections = new[]
    {
        new Section
        {
            Text = "ABC",
            Actions = new[]
            {
                new OpenUriAction
                {
                    Name = "OpenUriAction",
                    Type = ActionType.OpenUri,
                    Targets = new List<Target>
                    {
                        new Target
                        {
                            OS = TargetOs.Default,
                            Uri = "https://www.google.com"
                        }
                    }
                }
            }
        }
    }
};
```

And here's the generated JSON from `ToJson()`
```json
{
    "@type": "MessageCard",
    "@context": "http://schema.org/extensions",
    "originator": null,
    "summary": null,
    "themeColor": null,
    "title": "This is the title",
    "text": "Message text",
    "sections": [
        {
            "title": null,
            "startGroup": false,
            "activityTitle": null,
            "activitySubtitle": null,
            "activityText": null,
            "activityImage": null,
            "heroImage": null,
            "text": "ABC",
            "facts": null,
            "images": null,
            "potentialAction": [
                {
                    "targets": [
                        {
                            "os": "Default",
                            "uri": "https://www.google.com"
                        }
                    ],
                    "@type": 2,
                    "name": "OpenUriAction"
                }
            ]
        }
    ],
    "potentialAction": null
}
```

The issue here is that the button is not visible as the action type is being converted to `2`, instead of `"OpenUri"` as you can see in the below image (I've pasted the ToJson() output to [the message card playground](https://messagecardplayground.azurewebsites.net/))

![image](https://user-images.githubusercontent.com/926823/114338509-8b55fc80-9b96-11eb-9012-2ca36195ea03.png)

If I change the `2` to `"OpenUri"` in the message card playground site, it shows the button correctly
![image](https://user-images.githubusercontent.com/926823/114338628-c9532080-9b96-11eb-8ac3-1964343fd680.png)

Edit: Switching from 1.1.2 to 1.0.5 resolves my issues

**Comment by poychang** [comment]
@willl thanks for report this enum issue. It will be fix in 1.1.3.

**Comment by willl** [comment]
@poychang Thank you so much! Much appreciated.

I'll check the update out later
